### ISSUE TITLE
How do I change default merge target to develop, rather than master (for hub sync)?

### ISSUE THREAD

**Issue by Hubro** [github/hub#2349]
Sorry in advance if this has been answered somewhere else, but I couldn't find an answer in the README or the existing issues.

When I run `hub sync` I often get this warning:

    warning: 'feature/my-feature-name' was deleted on origin, but appears not merged into 'master'

However, in my projects we don't merge branches into `master`, we merge them into `develop`. Branches are only merged into `master` after a release.

Is it possible to configure `hub sync` to consider `develop` the target branch for merges rather than `master`?

**Comment by mislav** [comment]
Hub detects the default branch for a repository like this: `git symbolic-ref refs/remotes/origin/HEAD`

This will likely be "master" for your repository. You can change it to "develop":

```
git remote set-head origin develop
```

Note that this might affect other hub commands too:
* The default base for `hub pull-request` will now be "develop"

**Comment by Hubro** [comment]
Nice, thanks! Good to hear it was a quick fix.

Is this documented anywhere?

**Comment by mislav** [comment]
@hubro No. Perhaps someone could add a short section about default branches to both `pull-request.go` and `sync.go` help text?

**Comment by atwright147** [comment]
Which other commands might this affect?

**Comment by mislav** [comment]
@atwright147 None that I can think of

**Comment by Alsrroum** [comment]
web doc how to change it: https://docs.github.com/en/github/administering-a-repository/changing-the-default-branch
